modules/prepare-operator-upgrades.adoc

```diff
@@ -6,8 +6,14 @@
 = Preparing to upgrade
 
 [role="_abstract"]
-Before you upgrade {rh-rhacs-first} version, you must:
+Before you upgrade the {rh-rhacs-first} version, you must perform the following steps:
 
 * If you are upgrading from version 3.74, verify that you are running the latest patch release version of the {product-title-short} Operator 3.74.
 * Backup your existing Central database.
-* If the cluster you are upgrading contains the `SecuredCluster` custom resource (CR), change the collection method to `EBPF` or `CORE_BPF`.
+* If the `SecuredCluster` custom resource is present in the cluster, ensure the per node collection setting is set to EBPF by following these steps:
```

Comment:
Hm. The suggestion to set to `EBPF` seems to come from the past. I believe, `CORE_BPF` is aimed to be the default from 4.5.0.

Let's ask the @erthalion @robbycochran @Stringy to double-check here.
Preview is https://65388--ocpdocs-pr.netlify.app/openshift-acs/latest/upgrading/upgrade-operator#prepare-operator-upgrades_upgrade-operator

Reply:
`CORE_BPF` is already the default in 4.4.

Reply:
Thanks @JoukoVirtanen this is valuable info but it is still not enough for the question I asked. I asked if it is correct to ask users set collection method to EBPF on upgrading? We're talking about upgrade instructions to version 4.5.0.

Reply:
How is the upgrade procedure special in this context?

I see following options:

* A regular one-version upgrade from 4.4 to 4.5 . As Jouko has mentioned, `CORE_BPF` is already a default one in 4.4, and we recommend to not stick to `EBPF` if not strictly needed.
* An upgrade from older version where `CORE_BPF` wasn't a default collection method. This instruction still doesn't make sense, because the last upgrade to 4.5 will render it useless.

I assume it comes from the previous deprecation of "kernel modules" collection method (might be the case some manual intervention was needed back then to switch to `EBPF`). Even more, the instructions described here wont be enough to enforce `EBPF`, as the operator contains translation logic that will still make the cluster use `CORE_BPF`.

To summarize, I don't think ensuring `EBPF` is needed. Does it make sense to you?

Reply:
From what we discussed, both old content and new content appear to be invalid. Could you suggest something more correct, please?

I think we need to consider cases (remember we upgrade to 4.5.0 in this iteration)
1. Upgrading from pre-4.1.0 when KERNEL_MODULE was around.
2. Upgrading from [4.1.0, 4.4.0) when CORE_BPF was not available.
3. Upgrading from 4.4.0 or later.

In case of (1) the collection method MUST be switched to EBPF otherwise bad things happen but I think we wrote about it in docs as well, did not we? If yes, it seems the documentation becomes fragmented.

In cases (2) and (3), I understand it does not matter what collection method is because CORE_BPF will actually be used. In the case (3) it will be nice to tell to switch to CORE_BPF so that the config is consistent with the reality but this may confuse folks who want to force EBPF so I'm not sure if we should bother.

Reply:
If it would be up to me, I would remove instructions around changing the collection method altogether, and instead add a warning at the end of this page saying "if you upgrade from version 4.1.0 or below, make sure to the collection method is CORE_BPF" and explanation how to verify this and modify if needed. How does it soun?

Reply:
I think that this is obsolete after Jouko and I made changes regarding the collection method during the last release. See:

https://docs.openshift.com/acs/4.4/upgrading/upgrade-operator.html#change-collection-method_upgrade-operator

I'll change the language and add a link (cannot add a link directly in the text, but will add it under additional references - this is a limitation of our publishing system).

Reply:
Actually, the Change Collection Method section is directly after this "preparing" section, so...I don't think it needs a link in additional references.